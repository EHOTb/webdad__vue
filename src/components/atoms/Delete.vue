<template>
    <div :class="$style.delete">&#10006;</div>
</template>

<script>
export default {
    name: "Delete"
}
</script>

<style lang="scss" module>
@import "../../assets/styles/var.scss";

.delete {
    display: flex;
    justify-content: center;
    border: 0.15rem solid $delete;
    padding: 0.5rem 0.5rem 0.3rem;
    border-radius: 20%;
    color: $delete;
}

.delete:hover {
    box-shadow: 0.2rem 0.2rem 0.2rem $main-col;
    cursor: pointer;
    transition: 0.3s;
}
</style>