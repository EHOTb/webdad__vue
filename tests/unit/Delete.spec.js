import { mount } from "@vue/test-utils";
import Delete from "../../src/components/atoms/Delete";

describe("Компонент Delete", () => {
  it("Delete snapshot", () => {
    const wrapper = mount(Delete);
    expect(wrapper.element).toMatchSnapshot();
  });
  it("emit события Delete", () => {
    const wrapper = mount(Delete);
    wrapper.find("div").trigger("click");
    expect(wrapper.emitted("delete")).toBeTruthy();
  });
});
