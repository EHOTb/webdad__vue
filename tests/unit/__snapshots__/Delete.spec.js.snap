// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Delete Delete snapshot 1`] = `
<div
  class="delete"
>
  ✖
</div>
`;
